### ISSUE TITLE
LOTS more comments needed!!!

### ISSUE THREAD

**Issue by dabockster** [ColdSauce/renaissance#2]
I know you still need to write formal documentation, but I still have no idea what the code is supposed to be doing. Since it's Java, I expect to see psuedo code comments everywhere. Because of Java's nature, even when there are some block comments, I still can't make heads or tails of what the overall idea of each method is.

For example, I usually use double slash // comments throughout my code as an informal way of allowing both myself and others trace through methods and logic, as well as the more traditional JavaDoc comments.

**Comment by ColdSauce** [comment]
I follow the methodology that you should comment *why* you wrote a piece of code the way you did. Not *what* the code does. 

A person should be able to figure out what the code does if the code is written well enough. 

However, I understand that a lot of the code isn't as great as it could be and a lot of things aren't documented. I'll spend most of the day documenting and making the code a lot better.

Thanks!

**Comment by dabockster** [comment]
Thanks a lot. Its very hard to parse mentally in its current state.

**Comment by ColdSauce** [comment]
I have added as many comments as I view necessary and cleaned the code as much as possible to allow for anyone with Java experience to understand what is going on fairly easily.
